src/ios/platform.mm

```diff
@@ -35,8 +35,15 @@
   return error.localizedDescription;
 }
 
+static std::string s_default_realm_directory;
+
 namespace realm {
 
+void set_default_realm_file_directory(std::string dir)
+{
+    s_default_realm_directory = dir;
```

Comment:
Let me add that if you set the default directory, it will used. It is likely that it will give issues for iOS if you set the path outline your sandbox but it is already documented.